### PR TITLE
Update docs for updated citations/references policy (#702)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
@betterscientificsoftware/bssw-editorial-board 

* Added the agreed-to citations and references policy given in https://github.com/betterscientificsoftware/bssw.io/issues/702#issuecomment-787012416 and I updated the publishing checklist to mention `wikize_refs.py` ("702").  See [Citations/References](https://github.com/bartlettroscoe/bssw.io/blob/702-citations-policy/docs/pages/bssw/bssw_styling_originalarticles.md#citationsreferences), [Style/formatting Checks](https://github.com/bartlettroscoe/bssw.io/blob/702-citations-policy/docs/pages/bssw/bssw_content_publishing.md#styleformatting-checks), and [Pre-publishing Checks](https://github.com/bartlettroscoe/bssw.io/blob/702-citations-policy/docs/pages/bssw/bssw_content_publishing.md#pre-publishing-checks).

* Updated the document `WhenNotToWriteAutomatedTests.md` to match the new policy as a demonstration.

* Updated so Jekyll works on my machine (but comment out local stuff as it was).

* Other small changes related to this (see the individual commits for details).

I viewed the updated files both on my local machine using locally-installed Jekyll and also in the branch:

* https://github.com/bartlettroscoe/bssw.io/blob/702-citations-policy/

NOTE: I would be good to add a PR template with a checklist as well but that can be done in a follow-up PR.

### CODE REVIEW

File: docs/pages/bssw/bssw_content_publishing.md [hunk 1]
```diff
@@ -42,10 +42,13 @@ The following steps assume that the BSSw editorial team has already determined t
 * Hyperlinks validity.
 * Check for any prior versions of the same content. If the prior version is very outdated, consider replacing it with this newer content. Please judge on a case-by-case basis.
 * Past events of the same conferences/workshops/etc. are not deleted and just kept as archives.
+* If using [formal citations/references](bssw_styling_originalarticles.html#citationsreferences), use [`wikize_refs.py -i <base>.md`](https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy) to deal with these (see [here](https://github.com/betterscientificsoftware/bssw.io/blob/master/Articles/Blog/ReferencesInMarkdownHybridApproach.md) for details).
 
 ### Pre-publishing Checks
 * Finalize the category(ies), topic(s) and other parameters in metadata.
-* Check on preview site to confirm overall display ("publish: preview" in metadata).
-* Change "publish: yes" in metadata before merging into master.
+* Ensure to run [`wikize_refs.py -i <base>.md`](https://github.com/betterscientificsoftware/bssw.io/blob/master/Articles/Blog/ReferencesInMarkdownHybridApproach.md) and commit before previewing (if `wikize_refs.py` is being used).
```
Comment (markcmiller86) [inline docs/pages/bssw/bssw_content_publishing.md:49]: Ensure to run --> Be sure to run

or

Ensure to run [`wikize_refs.py -i <base>.md`] --> Ensure [`wikize_refs.py -i <base>,md`] is run

**Comment by bartlettroscoe** [comment]
FYI: I had to manually resolve a merge conflict with this PR branch to 'preview'.  See commit https://github.com/betterscientificsoftware/bssw.io/pull/750/commits/032b29fed329fcd090f5501968d85060e08bae49 for details.

File: docs/pages/bssw/bssw_styling_originalarticles.md [hunk 2]
```diff
@@ -70,6 +70,18 @@ This section has details of the author in a short paragraph. The content is free
...
+
+* Content that in the judgement of EB members and/or authors requires references to more fully support perhaps sensitive and/or controversial positions (e.g. the Covid19 article).
+
+The decision to *allow* or *require* references is one that should be agreed upon by the author and EB members prior to developing the content. When references are to be used, we require authors to use the less intrusive [reference links](https://www.markdownguide.org/basic-syntax#reference-style-links) ([full spec](https://github.github.com/gfm/#reference-link)) and to follow the guidelines described [here](https://github.com/betterscientificsoftware/bssw.io/blob/master/Articles/Blog/ReferencesInMarkdownHybridApproach.md) where the [`wikize_refs.py`](https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy) tool can be helpful.
+
 ## Existing Examples
 
 * A skeleton Markdown template for a blog article, which one can copy and customize is available at [this location](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/blob/master/Articles/Blog/BlogArticleSkeletonA.md)
```
Comment (markcmiller86) [inline docs/pages/bssw/bssw_styling_originalarticles.md:87]: Can we include an example and link to it that includes use of references? Maybe `WhenNotToWriteAutomatedTests.md` here and indicate its a good example of the use of citations/references.

**Review by markcmiller86 — CHANGES_REQUESTED** [review]

File: docs/pages/bssw/bssw_styling_originalarticles.md (hunk 2 above)
Comment (bartlettroscoe) [reply docs/pages/bssw/bssw_styling_originalarticles.md:87]: @markcmiller86, just be clear, should the link to `WhenNotToWriteAutomatedTests.md` as an example go into the section `Existing Examples` or the section on citations/references?

Comment (markcmiller86) [reply docs/pages/bssw/bssw_styling_originalarticles.md:87]: I think `Existing Examples` as I believe that to be a list of examples that demonstrate various useful aspects what is described here. But, each example alone is probably not sufficient and maybe a short line or phrase that goes with the example(s) as to *what* the example is good for demonstrating (thought I wouldn't expect you to do that for all the pre-existing examples there, at least do it for the new one you will add that demos citrations). Does that make sense?

Comment (bartlettroscoe) [reply docs/pages/bssw/bssw_styling_originalarticles.md:87]: @markcmiller86, the link to the example file is in commit https://github.com/betterscientificsoftware/bssw.io/pull/750/commits/f1970b0be8264c8e89cf2ac584f0d4ff2ddc0fc9.  

NOTE: This link points to this `*.md` file on the 'master' branch but the updated `*.md` file is on this PR branch so that link will only point to the updated example once this PR gets merged.

File: docs/pages/bssw/bssw_content_publishing.md (hunk 1 above)
Comment (bartlettroscoe) [reply docs/pages/bssw/bssw_content_publishing.md]: Thanks!  Addressed in commit https://github.com/betterscientificsoftware/bssw.io/pull/750/commits/a1eab3be92286d232789f159d147022d1b825950.

**Comment by bartlettroscoe** [comment]
@markcmiller86, I think I have addressed your suggestions above in recent commits pushed.  Can you please update your review?

**Review by markcmiller86 — APPROVED** [review]

**Comment by bartlettroscoe** [comment]
@betterscientificsoftware/bssw-editorial-board,

@markcmiller86 gave his review and feedback which I addressed.  Does anyone else want to review this material before it gets merged to 'master'?  We can always tweak it after it gets merged to 'master' as well.

**Comment by bartlettroscoe** [comment]
@rinkug said she wants to review and approve tomorrow.

File: docs/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/_site/
```
Comment (rinkug) [inline docs/.gitignore:1]: Is this file needed for this commit? I am accepting this for now.

File: docs/pages/bssw/bssw_content_publishing.md [hunk 4]
```diff
@@ -42,10 +42,13 @@ The following steps assume that the BSSw editorial team has already determined t
 * Hyperlinks validity.
 * Check for any prior versions of the same content. If the prior version is very outdated, consider replacing it with this newer content. Please judge on a case-by-case basis.
 * Past events of the same conferences/workshops/etc. are not deleted and just kept as archives.
+* If using [formal citations/references](bssw_styling_originalarticles.html#citationsreferences), use [`wikize_refs.py -i <base>.md`](https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy) to deal with these (see [here](https://github.com/betterscientificsoftware/bssw.io/blob/master/Articles/Blog/ReferencesInMarkdownHybridApproach.md) for details).
```
Comment (rinkug) [inline docs/pages/bssw/bssw_content_publishing.md:45]: Please remind me why are "ReferencesInMarkdown*" files sitting in blog directory? Is it because,  historically they were used for blog and put in that folder? Perhaps, we should move them to a different location, outside of content? Is NOW the right time to do it? Perhaps open another PR for this, if needed.

**Review by rinkug — APPROVED** [review]

File: docs/.gitignore (hunk 3 above)
Comment (bernhold) [reply docs/.gitignore:1]: The `_site` directory is build and populated by Jekyll's generation process.  With a basic Jekyll site hosted on GH pages, this is handled automatically behind the scenes.

If the Jekyll site becomes more sophisticated, the GH setup may not be able to build it automatically, in which case you need to put `_site` under version control and generate and commit it as it gets updated.  The most common scenario for this to become necessary is using Jekyll plugins that are not on GH pages's safe list.  It is possible to use GH Actions to build/"deploy" Jekyll sites, and there exist actions for this in the Marketplace.  (I haven't looked deeply enough to know what limitations they might have.)

Comment (bartlettroscoe) [reply docs/.gitignore:1]: > Is this file needed for this commit? I am accepting this for now.

That is a Git ignore for a generated directory of files.  That means that we **don't** want to commit those files.
